packages/client/src/vscode/apiWrapper.ts

```diff
@@ -0,0 +1,333 @@
+            });
+        }
+    }
+
+    protected async configureViewsServices() {
+        const viewServiceType = this.apiConfig.viewsConfig?.viewServiceType ?? 'EditorService';
+        if (this.apiConfig.$type === 'classic' && (viewServiceType === 'ViewsService' || viewServiceType === 'WorkspaceService')) {
+            throw new Error(`View Service Type "${viewServiceType}" cannot be used with classic configuration.`);
+        }
+
+        const envEnhanced = getEnhancedMonacoEnvironment();
+        if (this.apiConfig.viewsConfig?.viewServiceType === 'ViewsService') {
+            const getViewsServiceOverride = (await import('@codingame/monaco-vscode-views-service-override')).default;
+            mergeServices(this.apiConfig.serviceOverrides, {
+                ...getViewsServiceOverride(this.apiConfig.viewsConfig.openEditorFunc ?? useOpenEditorStub)
+            });
+            envEnhanced.viewServiceType = 'ViewsService';
+        } else if (this.apiConfig.viewsConfig?.viewServiceType === 'WorkspaceService') {
+            const getWorkbenchServiceOverride = (await import('@codingame/monaco-vscode-workbench-service-override')).default;
+            mergeServices(this.apiConfig.serviceOverrides, {
+                ...getWorkbenchServiceOverride()
+            });
+            envEnhanced.viewServiceType = 'WorkspaceService';
+        } else {
+            const getEditorServiceOverride = (await import('@codingame/monaco-vscode-editor-service-override')).default;
+            mergeServices(this.apiConfig.serviceOverrides, {
+                ...getEditorServiceOverride(this.apiConfig.viewsConfig?.openEditorFunc ?? useOpenEditorStub)
+            });
+            envEnhanced.viewServiceType = 'EditorService';
+        }
+    }
+
+    protected async applyViewsPostConfig() {
+        this.apiConfig.viewsConfig?.htmlAugmentationInstructions?.(this.apiConfig.htmlContainer);
+        await this.apiConfig.viewsConfig?.viewsInitFunc?.();
+    }
+
+    /**
+     * Adding the default workspace config if not provided
+     */
+    protected configureWorkspaceConfig() {
+        if (this.apiConfig.workspaceConfig === undefined) {
+            this.apiConfig.workspaceConfig = {
+                workspaceProvider: {
+                    trusted: true,
+                    workspace: {
+                        workspaceUri: vscode.Uri.file('/workspace.code-workspace')
+                    },
+                    async open() {
+                        window.open(window.location.href);
+                        return true;
+                    }
+                },
+            };
+        }
+    }
+
+    /**
+     * set the log-level via the development settings
+     */
+    protected configureDevLogLevel() {
+        const devLogLevel = this.apiConfig.workspaceConfig?.developmentOptions?.logLevel;
+        if (devLogLevel === undefined) {
+
+            // this needs to be done so complicated, because developmentOptions is read-only
```

Comment:
Can probably update the function header above to explain the process a bit more.

Reply:
I will update the function comment.